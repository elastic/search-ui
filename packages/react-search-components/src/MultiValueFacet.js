import PropTypes from "prop-types";
import React from "react";

import { ValueFacetOption, ValueFilterValue } from "./types";

function MultiValueFacet({
  label,
  onMoreClick,
  onRemove,
  onSelect,
  options,
  showMore,
  values
}) {
  return (
    <div className="sui-multi-value-facet sui-facet">
      <div className="sui-multi-value-facet__label">{label}</div>
      <div className="sui-multi-value-facet__options-list">
        {options.map(option => {
          const checked = values.includes(option.value);
          return (
            <label
              key={`${option.value}`}
              htmlFor={`example_facet_${label}${option.value}`}
              className="sui-multi-value-facet__option-label"
            >
              <div className="sui-multi-value-facet__option-input-wrapper">
                <input
                  id={`example_facet_${label}${option.value}`}
                  type="checkbox"
                  className="sui-multi-value-facet__checkbox"
                  checked={checked}
                  onChange={() =>
                    checked ? onRemove(option.value) : onSelect(option.value)
                  }
                />
                <span className="sui-multi-value-facet__input-text">
                  {option.value}
                </span>
              </div>
              <span className="sui-multi-value-facet__option-count">
                {option.count.toLocaleString("en")}
              </span>
            </label>
          );
        })}
      </div>
      {showMore && (
        <div className="sui-multi-value-facet__view-more" onClick={onMoreClick}>
          + More
        </div>
      )}
    </div>
  );
}

MultiValueFacet.propTypes = {
  label: PropTypes.string.isRequired,
  onMoreClick: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(ValueFacetOption).isRequired,
  showMore: PropTypes.bool.isRequired,
  values: PropTypes.arrayOf(ValueFilterValue).isRequired
};

export default MultiValueFacet;
