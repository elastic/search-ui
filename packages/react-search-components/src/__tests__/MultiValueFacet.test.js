import React from "react";
import MultiValueFacet from "../MultiValueFacet";
import { shallow } from "enzyme";

const params = {
  label: "A Facet",
  onMoreClick: jest.fn(),
  onRemove: jest.fn(),
  onSelect: jest.fn(),
  options: [
    {
      value: "fieldValue1",
      count: 10
    },
    {
      value: "fieldValue2",
      count: 5
    }
  ],
  showMore: true,
  values: ["fieldValue2"]
};

it("renders", () => {
  const wrapper = shallow(<MultiValueFacet {...params} />);
  expect(wrapper).toMatchSnapshot();
});

it("will render 'more' button if more param is true", () => {
  const wrapper = shallow(
    <MultiValueFacet
      {...{
        ...params,
        showMore: true
      }}
    />
  );
  expect(wrapper.find(".sui-multi-value-facet__view-more")).toHaveLength(1);
});

it("won't render 'more' button if more param is false", () => {
  const wrapper = shallow(
    <MultiValueFacet
      {...{
        ...params,
        showMore: false
      }}
    />
  );
  expect(wrapper.find(".sui-multi-value-facet__view-more")).toHaveLength(0);
});
