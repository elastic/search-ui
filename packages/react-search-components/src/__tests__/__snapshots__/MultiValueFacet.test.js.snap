// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sui-multi-value-facet sui-facet"
>
  <div
    className="sui-multi-value-facet__label"
  >
    A Facet
  </div>
  <div
    className="sui-multi-value-facet__options-list"
  >
    <label
      className="sui-multi-value-facet__option-label"
      htmlFor="example_facet_A FacetfieldValue1"
      key="fieldValue1"
    >
      <div
        className="sui-multi-value-facet__option-input-wrapper"
      >
        <input
          checked={false}
          className="sui-multi-value-facet__checkbox"
          id="example_facet_A FacetfieldValue1"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="sui-multi-value-facet__input-text"
        >
          fieldValue1
        </span>
      </div>
      <span
        className="sui-multi-value-facet__option-count"
      >
        10
      </span>
    </label>
    <label
      className="sui-multi-value-facet__option-label"
      htmlFor="example_facet_A FacetfieldValue2"
      key="fieldValue2"
    >
      <div
        className="sui-multi-value-facet__option-input-wrapper"
      >
        <input
          checked={true}
          className="sui-multi-value-facet__checkbox"
          id="example_facet_A FacetfieldValue2"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="sui-multi-value-facet__input-text"
        >
          fieldValue2
        </span>
      </div>
      <span
        className="sui-multi-value-facet__option-count"
      >
        5
      </span>
    </label>
  </div>
  <div
    className="sui-multi-value-facet__view-more"
    onClick={[MockFunction]}
  >
    + More
  </div>
</div>
`;
